### PR TITLE
fix(apigateway): allow multiple api keys to the same usage plan

### LINKED ISSUE #4860: [apigateway] Cannot call UsagePlan.addApiKey multiple times.
It appears to be impossible to add multiple API keys to a usage plan in the API Gateway module. Adding a single key works fine, but there is a construct naming conflict when adding multiple keys.

### Reproduction Steps

```typescript
// const api: apigateway.RestApi = new RestApi()....;

const plan = api.addUsagePlan('DefaultUsagePlan', {
    description: `Default usage plan for ${api.node.id}`,
    name: 'DefaultUsagePlan'
});
plan.addApiKey(api.addApiKey('ClientKey'));
plan.addApiKey(api.addApiKey('TestingKey'));
```

### Error Log

When adding a second API key, I get an error complaining about the reuse of a construct name in CDK.

```
There is already a Construct with name 'UsagePlanKeyResource' in UsagePlan [DefaultUsagePlan]

.../infrastructure/node_modules/@aws-cdk/core/lib/construct.ts:48
        throw new Error(`Validation failed with the following errors:\n  ${errorList}`);
              ^
Error: Validation failed with the following errors:
  [LambdaStack/BotAudioVideoApi] The REST API doesn't contain any methods
    at Function.synth (.../infrastructure/node_modules/@aws-cdk/core/lib/construct.ts:48:15)
    at App.synth (.../infrastructure/node_modules/@aws-cdk/core/lib/app.ts:142:36)
    at process.App.process.once (.../infrastructure/node_modules/@aws-cdk/core/lib/app.ts:121:45)
    at Object.onceWrapper (events.js:286:20)
    at process.emit (events.js:198:13)
    at process.EventEmitter.emit (domain.js:448:20)
    at process.emit (.../infrastructure/node_modules/source-map-support/source-map-support.js:465:21)
    at process.topLevelDomainCallback (domain.js:126:23)
```

### Environment

  - **CLI Version      :** 1.12.0, 1.15.0
  - **Framework Version:** 1.12.0, 1.15.0
  - **OS               :** MacOS
  - **Language         :** Typescript

### Other

I initially tried this using CDK v1.12.0. I have reproduced it in CDK v1.15.0.

There is a work-around using `CfnUsagePlanKey`:

```typescript
// const api: apigateway.RestApi = new RestApi()....;

const clientKey = api.addApiKey('ClientKey');
const testingKey = api.addApiKey('TestingKey')
const clientUsagePlanKey = new apiGateway.CfnUsagePlanKey(this, 'ClientUsagePlanKey', {
    keyId: clientKey.keyId,
    keyType: 'API_KEY',
    usagePlanId: plan.usagePlanId
});
const testingUsagePlanKey = new apiGateway.CfnUsagePlanKey(this, 'TestingUsagePlanKey', {
    keyId: testingKey.keyId,
    keyType: 'API_KEY',
    usagePlanId: plan.usagePlanId
});
```

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by nija-at**
fixes #4860

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigateway/lib/usage-plan.ts [hunk 1]
```diff
@@ -178,7 +178,7 @@ export class UsagePlan extends Resource {
    * @param apiKey
    */
   public addApiKey(apiKey: IApiKey): void {
-    new CfnUsagePlanKey(this, 'UsagePlanKeyResource', {
+    new CfnUsagePlanKey(this, `UsagePlanKey${apiKey.node.id}`, {
```
Comment (eladb) [inline packages/@aws-cdk/aws-apigateway/lib/usage-plan.ts]:
```suggestion
    const prefix = `UsagePlanKey`;
    const id = this.node.tryGetChild(prefix) ? `${prefix}:${apiKey.node.uniqueId}` : prefix;
    new CfnUsagePlan(this, id, {
```

File: packages/@aws-cdk/aws-apigateway/test/integ.restapi.expected.json [hunk 2]
```diff
@@ -602,7 +602,7 @@
         "UsagePlanName": "Basic"
       }
     },
-    "myapiUsagePlanUsagePlanKeyResource050D133F": {
+    "myapiUsagePlanUsagePlanKeyApiKeyF40DC38C": {
```
Comment (eladb) [inline packages/@aws-cdk/aws-apigateway/test/integ.restapi.expected.json]: We can avoid this

File: packages/@aws-cdk/aws-apigateway/lib/usage-plan.ts (hunk 1 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-apigateway/lib/usage-plan.ts]: Thanks. Will get this in.

Why do you suggest `uniqueId` over `id`? 
A scope-wide uniqueness is sufficient here, while, using `uniqueId` would make the physical id of this resource really long.

File: packages/@aws-cdk/aws-apigateway/lib/usage-plan.ts [hunk 3]
```diff
@@ -178,7 +178,9 @@ export class UsagePlan extends Resource {
    * @param apiKey
    */
   public addApiKey(apiKey: IApiKey): void {
-    new CfnUsagePlanKey(this, 'UsagePlanKeyResource', {
+    const prefix = 'UsagePlanKeyResource';
+    const id = this.node.tryFindChild(prefix) ? `${prefix}:${apiKey.node.id}` : prefix;
```
Comment (eladb) [inline packages/@aws-cdk/aws-apigateway/lib/usage-plan.ts]: - use `node.uniqueId` (`node.id` is not good enough)
- add a little comment explaining why you are only postfixing with the node ID in the 2nd child.

Comment (nija-at) [reply packages/@aws-cdk/aws-apigateway/lib/usage-plan.ts]: > use node.uniqueId (node.id is not good enough)

Please see my response to your previous comment on this asking why not.

Comment (eladb) [reply packages/@aws-cdk/aws-apigateway/lib/usage-plan.ts]: The api key is not defined in your scope and there is nothing that prevents users from passing in two different apiKeys with the same id defined in two different parent scopes.

**Pull request merged**